include/deal.II/dofs/dof_accessor.templates.h

```diff
@@ -2636,9 +2636,8 @@ DoFCellAccessor<dimension_, space_dimension_, level_dof_access>::get_fe() const
   const auto &fe = this->dof_handler->get_fe(active_fe_index());
 
   Assert(this->reference_cell() == fe.reference_cell(),
-         ExcMessage(
-           "The reference-cell type of the cell does not match the one of the "
-           "finite element!"));
+         internal::ExcNonMatchingReferenceCellTypes(this->reference_cell(),
+                                                    fe.reference_cell()));
```

Comment:
Let's not invent a whole new exception class for just one place. This is more easily done in the following way:
```suggestion
  ExcMessage ("The reference-cell type used on this cell (" + this->reference_cell().to_string() +
          ") does not match the reference-cell type of the finite element associated with this cell (" + 
          fe.reference_cell.to_string()
          + "). Did you accidentally use simplex elements on hypercube meshes (or the other way around), or "
          "are you using a mixed mesh and assigned a simplex element to a hypercube cell "
          "(or the other way around) via the "
          "active_fe_index?"));
```

Reply:
I would suggest to keep the class, since, if I remember correctly, there are other places, where it could be used.

Reply:
By performing a quick search I could at least replace one additional assert message. I'll have a deeper look if there are more relevant places. Then we can decide.

Reply:
If you need a new class, use `DeclExceptionMessage` instead of doing it by hand, though.